crates/deltalake-core/src/operations/constraints.rs

```diff
@@ -94,23 +95,29 @@ impl std::future::IntoFuture for ConstraintBuilder {
 
             if metadata.configuration.contains_key(&configuration_key) {
                 return Err(DeltaTableError::Generic(format!(
-                    "Constraint with name: {} already exists, expr: {}",
-                    name, expr
+                    "Constraint with name: {} already exists",
+                    name
                 )));
             }
 
             let state = this.state.unwrap_or_else(|| {
-                let session = SessionContext::new();
+                let session: SessionContext = DeltaSessionContext::default().into();
                 register_store(this.log_store.clone(), session.runtime_env());
                 session.state()
             });
 
-            // Checker built here with the one time constraint to check.
-            let checker = DeltaDataChecker::new_with_constraints(vec![Constraint::new("*", &expr)]);
             let scan = DeltaScanBuilder::new(&this.snapshot, this.log_store.clone(), &state)
                 .build()
                 .await?;
 
+            let schema = scan.schema().to_dfschema()?;
+            let expr = into_expr(expr, &schema, &state)?;
+            let expr_str = fmt_expr_to_sql(&expr)?;
```

Comment:
Is this going from SQL string expr to Datafusion Expr and then back to sql string expression?

Reply:
Yes exactly. 
If provided expression is already a Datafusion expression then `into_expr` will simply return it otherwise it will parse the string expression.
Either way if a String is provided or a proper DF expression it needs to be normalized somewhere.

Reply:
Which SQL dialect does Datafusion actually use?

Reply:
postgres ...  but the actual dialect expressions in the delta log should comply with is as of yet unspecified :).

Reply:
Should we put SQLglot in between on the python side then, so we can allow more flexibility there?

Reply:
We use the [generic dialect](https://github.com/delta-io/delta-rs/blob/4ece26d0da666b59df206515d24ab8797329b46f/crates/deltalake-core/src/delta_datafusion/expr.rs#L87) which in my mind is similar to postgres. Given we are only parsing expressions and not DML or DDL statements we should be fine.
https://docs.rs/sqlparser/latest/sqlparser/dialect/struct.GenericDialect.html

Reply:
> Should we put SQLglot in between on the python side then, so we can allow more flexibility there?

I'm indifferent to that feature. My only concern is if we start exposing operations to other languages (i.e delta-core) then we should have fairly consistent interfaces. 
A decision / documentation needs to be made on which sql dialect must be used when exposing expressions to the log. If I had to guess they would pick hive sql.